source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -308,6 +309,26 @@ void InstanceImpl::ThreadLocalActiveClient::onEvent(Network::ConnectionEvent eve
   }
 }
 
+// Inspired by the redis-cluster hashtagging algorithm
+// https://redis.io/topics/cluster-spec#keys-hash-tags
+std::string InstanceImpl::LbContextImpl::hashtag(const std::string& s, bool enabled) {
```

Comment:
Can you switch this to take an `absl::string_view` and return an `absl::string_view`? If we do that we can avoid any allocations I think?

Reply:
Tried implementing it using `absl::string_view`, however seems like std::string_view does not meet the requirements required for std::hash to work properly. Seems like the default constructor is deleted. :( 

```
bazel-out/darwin-fastbuild/bin/source/extensions/filters/network/redis_proxy/_virtual_includes/conn_pool_lib/extensions/filters/network/redis_proxy/conn_pool_impl.h:180:21: error: call to implicitly-deleted default constructor of 'std::hash<absl::string_view>'
        : hash_key_(std::hash<absl::string_view>()(hashtag(key, enabled_hashtagging))) {}
                    ^
```

Reply:
I would go ahead and construct an `std::string` just for computing the hash, and this can be fixed when we remove `std::hash` per the TODO in the next few lines.

Reply:
Sounds good 👍 I can work on that PR too, unless @HenryYYang wants to do it!

Reply:
Started the WIP PR [here](https://github.com/envoyproxy/envoy/pull/5766), will rebase everything here and add the missing release note.

Reply:
OK, well why don't we land this first? So update to string_view and then we can look at the other PR?